container-engine-lib/lib/backend_impls/docker/object_attributes_provider/enclave_object_attributes_provider.go

```diff
@@ -544,7 +546,7 @@ func (provider *dockerEnclaveObjectAttributesProviderImpl) getTraefikLabelsForEn
 			shortServiceUuid := uuid_generator.ShortenedUUIDString(serviceUuid)
 			servicePortStr := fmt.Sprintf("%s-%s-%d", shortEnclaveUuid, shortServiceUuid, portSpec.GetNumber())
 
-			labelKeyValuePairs[fmt.Sprintf("http.routers.%s.rule", servicePortStr)] = fmt.Sprintf("Host(`%d-%s-%s`)", portSpec.GetNumber(), shortServiceUuid, shortEnclaveUuid)
+			labelKeyValuePairs[fmt.Sprintf("http.routers.%s.rule", servicePortStr)] = fmt.Sprintf("HostRegexp(`{name:%d-%s-%s%s}`)", portSpec.GetNumber(), shortServiceUuid, shortEnclaveUuid, anyCharacterPrefixRegexToken)
```

Comment:
Do we need the group label `name`?

Reply:
Should we add the beginning and ending characters `^` and `$`:
`^81-3771c85af16a-65d2fb6d6732.*$`

Reply:
Should we add `-?` before `.*` so the enclave Id is delimited:
`^81-3771c85af16a-65d2fb6d6732-?.*$`

Reply:
@laurentluce yes, we need the group label name otherwise it will works only for the subdomain part by default.

Thanks for all the other suggestions I've incorporated all them in the matcher, I think the ^ and $ characters are not mandatory because the app add them if they are not present, but it's also works if we add them, so I'm adding them in order to have a better and cleaner regex.